File: servidor-spring/src/main/java/com/elspepegoteras/server/models/Usuari.java
package com.elspepegoteras.server.models;

import jakarta.persistence.*;

@Entity
public class Usuari {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(nullable = false)
    private String nom;

    @Column(unique = true)
    private String login;

    @Column(nullable = false)
    private String password;

    @Column(nullable = false)
    private String avatar;

    @Column(nullable = false)
    private int wins;

    @Column(nullable = false)
    private int games;

    //Constructor per defecte
    public Usuari() {}

    //Creació d'un usuari
    public Usuari(String nom, String login, String password) {
        setNom(nom);
        setLogin(login);
        setPassword(password);
        setAvatar(null);
        setWins(0);
        setGames(0);
    }

    //Recuperació d'un usuari
    public Usuari(Long id, String nom, String login, String password, String avatar, Integer wins, Integer games) {
        setId(id);
        setNom(nom);
        setLogin(login);
        setPassword(password);
        setAvatar(avatar);
        setWins(wins);
        setGames(games);
    }

    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }
    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getLogin() {
        return login;
    }
    public void setLogin(String login) {
        this.login = login;
    }

    public String getPassword() {
        return password;
    }
    public void setPassword(String password) {
        this.password = password;
    }

    public String getAvatar() {
        return avatar;
    }
    public void setAvatar(String avatar) {
        if (avatar == null || avatar.isEmpty()) {
            this.avatar = "avatar_home_1.png";
        } else {
            this.avatar = avatar;
        }
    }

    public Integer getWins() {
        return wins;
    }
    public void setWins(Integer wins) {
        this.wins = wins;
    }

    public Integer getGames() {
        return games;
    }
    public void setGames(Integer games) {
        this.games = games;
    }
}

File: client-wpf/RiskClient/RiskClient/Models/Usuari.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Media.Imaging;

namespace RiskClient.Models
{
    public class Usuari
    {
        public int Id { get; set; }
        public string Nom { get; set; }
        public string Login { get; set; }
        public string Password { get; set; }
        public string Avatar { get; set; }
        public int Wins { get; set; }
        public int Games { get; set; }

        public Usuari() { }

        public Usuari(int id, string nom, string login, string password, string avatar, int wins, int games)
        {
            Id = id;
            Nom = nom;
            Login = login;
            Password = password;
            Avatar = avatar;
            Wins = wins;
            Games = games;
        }

        public Usuari(string nom, string login, string password)
        {
            Nom = nom;
            Login = login;
            Password = password;
        }

        public Usuari(string login, string password)
        {
            Login = login;
            Password = password;
        }
    }
}
